components/inputs/input-date.js

```diff
@@ -212,12 +222,35 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		super.updated(changedProperties);
 
 		changedProperties.forEach((oldVal, prop) => {
-			if (prop === 'value') {
-				this._formattedValue = this.value ? formatISODateInUserCalDescriptor(this.value) : '';
+			if (prop === '_dateTimeDescriptor' || prop === 'value') {
+				this._setFormattedValue();
 			}
 		});
 	}
 
+	_getContentWidth() {
+		const text = document.createElement('div');
+		document.body.appendChild(text);
+		text.style.fontSize = '0.8rem';
+		text.style.width = 'auto';
+		text.style.position = 'absolute';
+
+		// in some languages (e.g., fr) placeholderWidth is bigger, in others (e.g., zh) contentWidth is bigger
+		text.innerHTML = (this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase();
+		const placeholderWidth = text.clientWidth;
+		text.innerHTML = formatISODateInUserCalDescriptor('2020-12-20');
+		const contentWidth = text.clientWidth;
```

Comment:
Definitely hacky. I can't think of a way that would accomplish the same thing and be as accurate. One idea I had initially would be to use the font-size & letter-spacing to come up with a formula that for a given number of characters you could guess a width. It incorrectly assumes that all characters are the same width though.

One optimization to this might be to only calculate it once for the first date-input... since having many of them on the page would add a bunch of divs, then remove them.

Reply:
The optimization would probably usually be okay, but we could run into problems if a later one is using the `empty-state-text` and it's wider